CMSIS-OS/ChibiOS/I2M_OXYGEN_NF/target_windows_devices_serialcommunication_config.cpp

```diff
@@ -0,0 +1,78 @@
+//
+// Copyright (c) 2017 The nanoFramework project contributors
+// See LICENSE file in the project root for full license information.
+//
+
+#include "win_dev_serial_native.h"
+
+///////////
+// UART1 //
+///////////
+
+// pin configuration for UART1
+// port: GPIOA
+// TX pin: is GPIOA_9
+// RX pin: is GPIOA_10
+// GPIO alternate pin function is 7 (see "Table 9. Alternate function mapping" in STM32F411xC and STM32F411xE datasheet)
+UART_CONFIG_PINS(1, GPIOA, 9, 10, 7)
+
+// buffers size
+// tx buffer size: 256 bytes
+#define UART1_TX_SIZE  256
+// rx buffer size: 256 bytes
+#define UART1_RX_SIZE  256
+
+// buffers
+// buffers that are R/W by DMA are recommended to be aligned with 32 bytes cache page size boundary
+// because of issues with cache coherency and DMA (this is particularly important with Cortex-M7 because of cache)
+#if defined(__GNUC__)
+__attribute__((aligned (32)))
+#endif
+uint8_t Uart1_TxBuffer[UART1_TX_SIZE];
+#if defined(__GNUC__)
+__attribute__((aligned (32)))
+#endif
+uint8_t Uart1_RxBuffer[UART1_RX_SIZE];
+
+// initialization for UART1
+UART_INIT(1, UART1_TX_SIZE, UART1_RX_SIZE)
+
+// un-initialization for UART1
+UART_UNINIT(1)
+
+
+///////////
+// UART2 //
+///////////
+
+// pin configuration for UART2
+// port: GPIOA
+// TX pin: is GPIOA_2
+// RX pin: is GPIOA_3
+// GPIO alternate pin function is 7 (see "Table 9. Alternate function mapping" in STM32F411xC and STM32F411xE datasheet)
+UART_CONFIG_PINS(2, GPIOA, 2, 3, 7)
+
+// buffers size
+// tx buffer size: 1024 bytes
+#define UART2_TX_SIZE  1024
+// rx buffer size: 1024 bytes
+#define UART2_RX_SIZE  1024
+
+// buffers
+// buffers that are R/W by DMA are recommended to be aligned with 32 bytes cache page size boundary
+// because of issues with cache coherency and DMA (this is particularly important with Cortex-M7 because of cache)
+#if defined(__GNUC__)
+__attribute__((aligned (32)))
+#endif
+uint8_t Uart2_TxBuffer[UART2_TX_SIZE];
+#if defined(__GNUC__)
+__attribute__((aligned (32)))
+#endif
+uint8_t Uart2_RxBuffer[UART2_RX_SIZE];
+
+// initialization for UART2
+UART_INIT(2, UART2_TX_SIZE, UART2_RX_SIZE)
```

Comment:
This has suffered a slight change to allow Gpio from different ports, please update or it won't work with the latest version.

Reply:
But what is not correct with one ?

Reply:
My bad, hit the wrong line... It's OK.

Reply:
It looks like the appveyor is using the old script, since F4-DISCOVERY is still mentioned ?